engine/src/main/java/org/hibernate/validator/internal/metadata/core/BuiltinConstraint.java

```diff
@@ -0,0 +1,139 @@
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_EAN("org.hibernate.validator.constraints.EAN", Arrays.asList( ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_MOD10_CHECK ) ),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_NOT_BLANK("org.hibernate.validator.constraints.NotBlank",
+			Arrays.asList( JAVAX_VALIDATION_CONSTRAINTS_NOT_NULL ) ),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_NOT_EMPTY("org.hibernate.validator.constraints.NotEmpty",
+			Arrays.asList( JAVAX_VALIDATION_CONSTRAINTS_NOT_NULL, JAVAX_VALIDATION_CONSTRAINTS_SIZE )),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_PARAMETER_SCRIPT_ASSERT("org.hibernate.validator.constraints.ParameterScriptAssert"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_RANGE("org.hibernate.validator.constraints.Range",
+			Arrays.asList( JAVAX_VALIDATION_CONSTRAINTS_MIN, JAVAX_VALIDATION_CONSTRAINTS_MAX )),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_SAFE_HTML("org.hibernate.validator.constraints.SafeHtml"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_SCRIPT_ASSERT("org.hibernate.validator.constraints.ScriptAssert"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_URL("org.hibernate.validator.constraints.URL",
+			Arrays.asList( JAVAX_VALIDATION_CONSTRAINTS_PATTERN )),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_UNIQUE_ELEMENTS("org.hibernate.validator.constraints.UniqueElements"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_BR_CNPJ("org.hibernate.validator.constraints.br.CNPJ",
+			Arrays.asList( JAVAX_VALIDATION_CONSTRAINTS_PATTERN )),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_BR_CPF("org.hibernate.validator.constraints.br.CPF",
+			Arrays.asList( JAVAX_VALIDATION_CONSTRAINTS_PATTERN )),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_BR_TITULO_ELEITORAL("org.hibernate.validator.constraints.br.TituloEleitoral",
+			Arrays.asList( JAVAX_VALIDATION_CONSTRAINTS_PATTERN, ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_MOD11_CHECK ) ),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_PL_NIP("org.hibernate.validator.constraints.pl.NIP"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_PL_PESEL("org.hibernate.validator.constraints.pl.PESEL"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_PL_REGON("org.hibernate.validator.constraints.pl.REGON"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_TIME_DURATION_MAX("org.hibernate.validator.constraints.time.DurationMax"),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_TIME_DURATION_MIN("org.hibernate.validator.constraints.time.DurationMin");
+
+	private static final Map<String, Set<BuiltinConstraint>> CONSTRAINT_MAPPING;
+
+	static {
+		CONSTRAINT_MAPPING = new HashMap<>();
+		for ( BuiltinConstraint constraint : values() ) {
+			if ( constraint.composingConstraints.isEmpty() ) {
+				CONSTRAINT_MAPPING.put( constraint.constraint, Collections.singleton( constraint ) );
+			}
+			else {
+				Set<BuiltinConstraint> constraints = new HashSet<>();
+				constraints.add( constraint );
+				constraints.addAll( constraint.composingConstraints );
+				CONSTRAINT_MAPPING.put( constraint.constraint, constraints );
+			}
+		}
+	}
+
+	private String constraint;
+	private List<BuiltinConstraint> composingConstraints;
```

Comment:
`requiredConstraints` or `constraintDependencies` or something like that? Feel free to ignore, but I had a hard time wrapping my head around this, so a more precise name might help.

Reply:
This is the technical term in HV but yeah I also thought about changing it to something along the lines of what you suggest so let's do it.